Modules/Backup-VCSA/Backup-VCSA.psm1

```diff
@@ -1,204 +1,462 @@
+	.EXAMPLE
+		The Below Create a schedule on Monday @11:30pm to FTP location 10.1.1.10:/vcsabackup/vcenter01
+		and keep 4 backups with a Encryption Passowrd of "VMw@re123"
+		
+		$location = "ftp://10.1.1.10/vcsabackup/vcenter01"
+		$LocationUser = "admin"
+		[VMware.VimAutomation.Cis.Core.Types.V1.Secret]$locationPassword = "VMw@re123"
+		$BHour = 23
+		$BMin = 30
+		$BDays = @("Monday")
+		$MaxCount = 4
+		[VMware.VimAutomation.Cis.Core.Types.V1.Secret]$BackupPassword = "VMw@re123"
+		
+		PS C:\> Set-VCSASchedule -Location $location -LocationUser $LocationUser -LocationPassword $locationPassword -BackupHour $BHour -BackupMinute $BMin -backupDays $BDays -MaxCount $MaxCount -BackupPassword $BackupPassword
+	.EXAMPLE
+		The Below Create a schedule on Sunday & Wednesday @5:15am
+		to NFS location 10.1.1.10:/vcsabackup/vcenter01
+		keep 10 backups with a Encryption Passowrd of "VMw@re123"
+		with Event Data included (Seat) and will delete any existing schedule.
+		
+		$location = "nfs://10.1.1.10/vcsabackup/vcenter01"
+		$LocationUser = "admin"
+		[VMware.VimAutomation.Cis.Core.Types.V1.Secret]$locationPassword = "VMw@re123"
+		$BHour = 5
+		$BMin = 15
+		$BDays = @("Sunday", "Monday")
+		$MaxCount = 10
+		[VMware.VimAutomation.Cis.Core.Types.V1.Secret]$BackupPassword = "VMw@re123"
+		
+		PS C:\> Set-VCSASchedule -IncludeSeat -force -Location $location -LocationUser $LocationUser -LocationPassword $locationPassword -BackupHour $BHour -BackupMinute $BMin -backupDays $BDays -MaxCount $MaxCount -BackupPassword $BackupPassword
+	.NOTES
+		Credit goes to @AlanRenouf & @vBrianGraf for sharing the base of this function.
+		You must be connected to the CisService for this to work, if you are not connected, the function will prompt you for your credentials
+#>
+	[CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Medium')]
+	param (
+		[Parameter(Mandatory=$true)]$Location,
+		[Parameter(Mandatory=$true)]$LocationUser,
+		[Parameter(Mandatory=$true)][VMware.VimAutomation.Cis.Core.Types.V1.Secret]$LocationPassword,
+		[Parameter(Mandatory=$true)][VMware.VimAutomation.Cis.Core.Types.V1.Secret]$BackupPassword,
+		[Parameter(Mandatory=$true)][ValidateRange(0,23)]$BackupHour,
+		[Parameter(Mandatory=$true)][ValidateRange(0,59)]$BackupMinute,
+		[Parameter(Mandatory=$true)][ValidateSet('Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday')][Array]$backupDays = $null,
+		[Parameter(Mandatory=$true)][Integer]$MaxCount,
+		[Parameter(Mandatory=$false)]$BackupID = "default",
+		[Parameter(Mandatory=$false)][switch]$IncludeSeat,
+		[Parameter(Mandatory=$false)][switch]$force
+	)
+	Begin {
+		if (!($global:DefaultCisServers)){ 
+			Add-Type -Assembly System.Windows.Forms
+			[System.Windows.Forms.MessageBox]::Show("It appears you have not created a connection to the CisServer. You will now be prompted to enter your vCenter credentials to continue" , "Connect to CisServer") | out-null
```

Comment:
I don't recommend to use message box. That assumes script is going to be used interactively and in GUI environment. Have in mind script could run in a background job, in non-interactive mode, over SSH, or over PowerShell remoting. 

My suggestion here is just to throw an error and request user to open connection and run the function again
throw "<message>".

Another option would be to add additional parameter $CisServer and if it is not specified it can be initialized with $global:DefaultCISServers. If it remains null afterwards throw error.